packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator.ClientModel/test/Providers/Definitions/ModelReaderWriterContextDefinitionTests.cs

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.TypeSpec.Generator.ClientModel.Providers;
+using Microsoft.TypeSpec.Generator.ClientModel.Tests;
+using Microsoft.TypeSpec.Generator.Input;
+using Microsoft.TypeSpec.Generator.Primitives;
+using Microsoft.TypeSpec.Generator.Providers;
+using Microsoft.TypeSpec.Generator.Tests.Common;
+using NUnit.Framework;
+
+namespace Microsoft.TypeSpec.Generator.ClientModel.Tests.Providers.Definitions
+{
+    public class ModelReaderWriterContextDefinitionTests
+    {
+        [Test]
+        public void ValidateModelReaderWriterContextIsGenerated()
+        {
+            MockHelpers.LoadMockGenerator();
+
+            var contextDefinition = new ModelReaderWriterContextDefinition();
+            
+            Assert.IsNotNull(contextDefinition);
+            Assert.IsNotNull(contextDefinition.Name);
+            Assert.IsTrue(contextDefinition.Name.EndsWith("Context"));
+            Assert.IsNotNull(contextDefinition.DeclarationModifiers);
+            Assert.IsTrue(contextDefinition.DeclarationModifiers.HasFlag(TypeSignatureModifiers.Public));
+            Assert.IsTrue(contextDefinition.DeclarationModifiers.HasFlag(TypeSignatureModifiers.Partial));
+            Assert.IsTrue(contextDefinition.DeclarationModifiers.HasFlag(TypeSignatureModifiers.Class));
+            Assert.IsNotNull(contextDefinition.Implements);
+            Assert.IsTrue(contextDefinition.Implements.Contains(typeof(ModelReaderWriterContext)));
+        }
+
+        [Test]
+        public void ValidateModelReaderWriterBuildableAttributesAreGenerated()
+        {
+            var mockGenerator = MockHelpers.LoadMockGenerator(
+                inputModels: () => new List<InputModelType>
+                {
+                    InputFactory.Model("TestModel", properties: new[]
+                    {
+                        InputFactory.Property("StringProperty", InputPrimitiveType.String),
+                        InputFactory.Property("IntProperty", InputPrimitiveType.Int32)
+                    })
+                });
+
+            var contextDefinition = new ModelReaderWriterContextDefinition();
+            var attributes = contextDefinition.Attributes;
+            
+            Assert.IsNotNull(attributes);
+            Assert.IsTrue(attributes.Count > 0);
+            
+            // Check that at least one ModelReaderWriterBuildableAttribute exists
+            var buildableAttributes = attributes.Where(a => a.Type.IsFrameworkType && a.Type.FrameworkType == typeof(ModelReaderWriterBuildableAttribute));
+            Assert.IsTrue(buildableAttributes.Any(), "At least one ModelReaderWriterBuildableAttribute should be generated");
+        }
+
+        [Test]
+        public void ValidateModelReaderWriterBuildableAttributesIncludeNestedModels()
```

Comment:
This test is implemented to simulated a nested model that is defined in the currently library being generated since both parentModel and nestedModel are being passed to inputModels mock.

We want another test which simulates the nestedModel is in dependency library which means it will be a csharptype from a framework type since you want have the model provider for it.  We still should get 2 buildable attributes.